src/profiling/meter_manager.cpp

```diff
@@ -142,6 +159,42 @@ void save_to_file(const meter_manager& manager, const std::string& name) {
     }
 }
 
+// Print easy to read report of meters so a stream.
+void print(const meter_manager& manager, std::ostream& o) {
```

Comment:
`util::print` is a very generic name for a function like this, and also conflates the text representation (a purely local operation) with the global communication rank test. I'd prefer that firstly the rank test be removed, and made the responsibility of the caller, and secondly that this be made an `operator<<` overload, or a method on the report struct, or at least called `pretty_print`.